tabbycat/checkins/management/commands/manageidentifiers.py

```diff
@@ -14,42 +14,41 @@ class Command(TournamentCommand):
     help = "Generates or deletes check-in identifiers"
 
     def add_arguments(self, parser):
-        subparsers = parser.add_subparsers(dest="subcommand", parser_class=ArgumentParser,
-              metavar="{generate,delete}")
+        parent = ArgumentParser(add_help=False)
+        parent.add_argument('-m', '--model', type=str, choices=['person', 'venue'], default='person',
+            help="Which model to use to create identifiers")
+
+        subparsers = parser.add_subparsers(dest="subcommand", parser_class=ArgumentParser)
         subparsers.required = True
 
-        generate = subparsers.add_parser("generate")
+        generate = subparsers.add_parser("generate", parents=[parent])
         super().add_arguments(generate)
         generate.add_argument('-O', '--overwrite', action="store_true", default=False,
             help="Overwrite existing URL keys")
-        generate.add_argument('-m', '--model', type=str, choices=['person', 'venue'], default='person',
-            help="Which model to use to create identifiers")
 
-        delete = subparsers.add_parser("delete")
+        delete = subparsers.add_parser("delete", parents=[parent])
         super().add_arguments(delete)
-        delete.add_argument('-m', '--model', type=str, choices=['person', 'venue'], default='person',
-            help="Which model to use to create identifiers")
 
     def handle_tournament(self, tournament, **options):
-        model = {
-            'person': Person,
-            'venue': Venue,
-        }[options['model']]
-        related_model = model.checkin_identifier.related.related_model
-
-        model_query = model.objects.all()
-        if options['model'] == 'venue':
-            model_query = model_query.filter(tournament=tournament)
-        else:
-            model_query = model_query.filter(Q(adjudicator__tournament=tournament) | Q(speaker__team__tournament=tournament))
-
-        if options['subcommand'] == "generate" and options['overwrite'] or options['subcommand'] == "delete":
-            self.stdout.write("Deleting all check-in identifiers for {0:s}...".format(options['model']))
-            delete_identifiers(model_query)
+        if options['model'] == 'person':
+            queryset = Person.objects.filter(
+                Q(adjudicator__tournament=tournament) | Q(speaker__team__tournament=tournament)
+            )
+            identifier_model = PersonIdentifier
```

Comment:
What's wrong with finding the related model through attributes?

Reply:
Oh, it just seems unnecessary given that it's known and we only deal with two models, and it makes the code more opaque. Especially in management commands, I'd like the code to be straightforward.

I may not always have lived up to this standard in other code, for many reasons (e.g. maybe something else seemed like a good idea at the time but doesn't now, or maybe it _was_ a good idea at the time but other things changed, or maybe it's just a borderline case).